app/models/user_helpers.rb

```diff
@@ -0,0 +1,29 @@
+module UserHelper
+
+  def registered_user?
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:current_user] = @user
```

Comment:
I would recommend against storing this back in the session. A `current_user` helper method would be more accessible and readable.